server/chat/backend/agent/tools/cloud_exec_tool.py

```
@@ -550,6 +564,11 @@ def setup_gcp_environment_isolated(user_id: str, selected_project_id: str | None
         logger.info("GCP isolated environment configured (%s)", auth_method)
         logger.info("TIME: setup_gcp_environment_isolated completed in %.2fs", time.perf_counter() - fn_start)
 
+        # Safety-net: register an atexit handler so the tmpdir is always
+        # removed even if the caller forgets to clean it up.  This covers
+        # every call-site, not just discovery_service.py.
+        atexit.register(_shutil.rmtree, cloudsdk_config_dir, True)
```

Ruff (0.15.12)

[warning] Boolean positional value in function call

(FBT003)